enhancements/network/dpu/ocp-on-dpu.md

```diff
@@ -0,0 +1,260 @@
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+See [DPU Support in OCP (Overview)]. This document describes the
+overall architecture, installation, and upgrades.
+
+[DPU Support in OCP (Overview)]: ./overview.md
+
+## Motivation
+
+See Overview doc.
+
+### Goals
+
+- Create a "Developer Preview" version of the feature for OCP 4.10:
+
+  - This will be difficult to install as there will be no explicit
+    support anywhere for installing the infra and tenant clusters
+    as a unit.
+
+  - Upgrades to the infra cluster will result in network outages in
+    the tenant cluster, as there will be no coordination between
+    infra cluster and tenant cluster reboots.
+
+  - The infra cluster will require separate ARM master nodes. We may
+    support running these as VMs on an x86 host.
+
+  - There will be no support for running customer workloads of any
+    sort in the infra cluster.
+
+  - It will not necessarily be possible to upgrade from Dev Preview
+    to Tech Preview without reinstalling one or both clusters.
+
+- Create further Tech Preview and GA versions of the feature with
+  improved functionality.
+
+### Non-Goals
+
+- ...
+
+## Proposal
+
+### User Stories
+
+See Overview doc.
+
+### Risks and Mitigations
+
+#### Hardware Availability
+
+We are currently developing using pre-GA BlueField-2 hardware, with a
+plan based on official descriptions of how the final GA BlueField-2
+hardware will work. So far (presumably due to the standard global chip
+supply chain problems?) we have been unable to get GA cards (either
+for free or by paying for them). If we are unable to get GA cards
+soon, it may not be possible to ship the Dev Preview as designed for
+OCP 4.10.
+
+#### OCP on ARM
+
+Some parts of OCP-on-bare-metal are not currently available for ARM.
+In particular, I think we don't have "ironic" images, which are needed
+for an IPI install.
+
+## Design Details
+
+### Design Details - Dev Preview
+
+The initial dev preview will be a very limited implementation, to
+demonstrate the overall architecture.
+
+#### Installation
+
+Installing a cluster with BF-2 cards is potentially tricky, since the
+cards can be configured in multiple modes, and in some modes, may not
+allow their host to access the network by default.
+
+To install the cluster then, it will be necessary to:
+
+- Install the tenant cluster masters and at least one non-BF-2 worker
+  node.
+
+- Create a MachineConfigPool in the tenant cluster for the BF-2
+  workers, and update the Cluster Network Operator configuration to
+  indicate that this pool is for BF-2 workers.
+
+- Power on, but don't yet install, the BF-2 hosts in the tenant
+  cluster (so that the BF-2s themselves will be powered on).
+
+- Install the infra cluster.
```

Comment:
Currently, both clusters share the same VLAN for the API network. We disconnected the DHCP server in each cluster manually during the cluster installation. And we do some manual IP assignments. I think we need to consider how to resolve this in the dev-preview.

Reply:
"disconnected the DHCP server" seems more like a problem with our particular network setup than a problem with the feature. (Customers will normally not try to install OCP on a network that has a DHCP server that they don't want to use.)

But I agree about VLANs, etc

Reply:
Currently, the infra and tenant clusters are in two different subnets, but using the same interface (hence the temporary disconnect of DHCP server). Discussion last week on whether it should be one subnet or not. Not sure if we came to a conclusion.